beamer/agent/middleware.py

```diff
@@ -282,3 +282,27 @@ def rate_limiter(
 ) -> Callable[[RPCEndpoint, Any], RPCResponse]:
     state = _RateLimiterState()
     return functools.partial(_rate_limiter, make_request=make_request, w3=w3, state=state)
+
+
+def base_fee_setter(
+    make_request: _MakeRequest, _w3: Web3
+) -> Callable[[RPCEndpoint, Any], RPCResponse]:
+
+    current_fees = make_request(RPCEndpoint("eth_feeHistory"), ["0x1", "latest", [10, 100]])[
+        "result"
+    ]
+    base_fee = max(current_fees["baseFeePerGas"])
+    max_fee = 2 * base_fee
+    priority_fee = make_request(RPCEndpoint("eth_maxPriorityFeePerGas"), [])["result"]
```

Comment:
Please be aware that `make_request` may fail so we need to check whether it was successful.